src/fides/api/service/connectors/sql_connector.py

```diff
@@ -129,12 +130,23 @@ def test_connection(self) -> Optional[ConnectionTestStatus]:
 
     def retrieve_data(
         self,
-        node: TraversalNode,
+        node: ExecutionNode,
         policy: Policy,
         privacy_request: PrivacyRequest,
+        request_task: RequestTask,
         input_data: Dict[str, List[Any]],
     ) -> List[Row]:
         """Retrieve sql data"""
+        # TODO REMOVE THIS IS FOR LOCAL TESTING FAILURES ON NODE
+        # if node.address.collection == "payment_card":
+        #     raise Exception()
+        # TODO REMOVE THIS IS FOR LOCAL CALLBACK TESTING
+        # if node.address.collection == "payment_card":
+        #     if request_task.callback_succeeded:
+        #         return []
+        #     # Send request and await for results. This will put the task
+        #     # in a paused state
+        #     raise AwaitingTaskCallback()
```

Comment:
Removed in https://github.com/ethyca/fides/pull/4760/files, but this POC illustrates raising an AwaitingTaskCallback exception for async connectors